### PR TITLE
(For 0.18.x) Expand expiry policy for HPP payment method without completition

### PR DESCRIPTION
**Pull request by daliwei**
Add a new property for HPP payments without completion actions.

For Hpp payments with completion action, it will use the PendingPaymentExpiry property. 

It also correct a bug that 3DS payment expiry policy was mistakenly used for Hpp pending as well (since both have a RedirectShopper result).

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 1]
```diff
@@ -774,6 +771,22 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
...
+            dao.updateResponse(kbTransactionId, properties, tenantId);
+        } catch (final SQLException e) {
+            throw new PaymentPluginApiException("SQL exception when updating response", e);
+        }
+    }
+
+    private AdyenResponsesRecord fetchResponseIfExist(final UUID kbPaymentId, final UUID tenantId) throws PaymentPluginApiException {
+        try {
+            return dao.getResponses(kbPaymentId, tenantId).stream().findFirst().orElse(null);
+        } catch (SQLException e) {
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Could be `final`.

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 2]
```diff
@@ -774,6 +771,22 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
...
+        try {
+            dao.updateResponse(kbTransactionId, properties, tenantId);
+        } catch (final SQLException e) {
+            throw new PaymentPluginApiException("SQL exception when updating response", e);
+        }
+    }
+
+    private AdyenResponsesRecord fetchResponseIfExist(final UUID kbPaymentId, final UUID tenantId) throws PaymentPluginApiException {
+        try {
+            return dao.getResponses(kbPaymentId, tenantId).stream().findFirst().orElse(null);
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Could you use `getSuccessfulAuthorizationResponse` instead (small perf gain as `limit 1` is done at the DB level)?

File: src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java [hunk 3]
```diff
@@ -136,6 +139,7 @@ public AdyenConfigProperties(final Properties properties, final String currentRe
 
         this.pendingPaymentExpirationPeriod = readPendingExpirationProperty(properties);
         this.pending3DsPaymentExpirationPeriod = read3DsPendingExpirationProperty(properties);
+        this.pendingHppPaymentWithouCompletionExpirationPeriod = readPendingHppPaymentWithouCompletionExpirationPeriod(properties);
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java]: `Withou` -> `Without`

File: src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java [hunk 4]
```diff
@@ -200,6 +204,17 @@ public AdyenConfigProperties(final Properties properties, final String currentRe
         }
     }
 
+    private Period readPendingHppPaymentWithouCompletionExpirationPeriod(final Properties properties) {
+        final String value = properties.getProperty(PROPERTY_PREFIX + "pendingHppPaymentWithoutCompletionExpirationPeriod");
+        if (value != null) {
+            try {
+                return Period.parse(value);
+            } catch (IllegalArgumentException e) { /* Ignore */ }
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java]: Nit: `final` (note that if you import the per project settings from `.idea/`, this should be taken care of for you).

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApiHPP.java [hunk 5]
```diff
@@ -96,6 +96,73 @@ public void testHPPWithPendingAuthorization() throws Exception {
         verifyPayment(TransactionType.AUTHORIZE);
     }
 
+    @Test(groups = "integration")
+    public void testAuthorizeAndExpireHppWithoutCompletition() throws Exception {
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApiHPP.java]: `Completition` -> `Completion`?

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApiHPP.java [hunk 6]
```diff
@@ -96,6 +96,73 @@ public void testHPPWithPendingAuthorization() throws Exception {
...
+                                                                          ImmutableList.<PluginProperty>of(),
+                                                                          context);
+        assertEquals(expiredPaymentTransactions.size(), 1);
+        final PaymentTransactionInfoPlugin canceledTransaction = expiredPaymentTransactions.get(0);
+        assertEquals(canceledTransaction.getTransactionType(), TransactionType.AUTHORIZE);
+        assertEquals(canceledTransaction.getStatus(), PaymentPluginStatus.CANCELED);
+    }
+
+    @Test(groups = "integration")
+    public void testAuthorizeAndExpireHppWithompletition() throws Exception {
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApiHPP.java]: `Withompletition` -> `WithCompletion`

File: src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java [hunk 7]
```diff
@@ -425,4 +440,8 @@ private synchronized void refillMap(final Map<String, String> map, final String
             }
         }
     }
+
+    public Period getPendingHppPaymentWithouCompletionExpirationPeriod() {
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java]: `Withou` -> `Without`

**Comment by daliwei** [comment]
@pierre Comments addressed in 2a1a160

**Comment by pierre** [comment]
0.5.12 has been released.

**Pull request merged**